script.js

```diff
@@ -0,0 +1,746 @@
+                        }
+                    });
+                }
+            });
+
+            // == restore regions array
+            app.display.regionsArray = ["AL", "AR", "DC", "MO", "PG"];
+            console.log("  app.state: ", app.state);
+        }
+    },
+
+    // ======= laneAjaxQueue =======
+    laneAjaxQueue: function(url, region) {
+        console.log("laneAjaxQueue");
+        var laneData = null;
+
+        $.ajax({
+            url: url,
+            method: "GET",
+            dataType: "text"
+        }).done(function(jsonData){
+            console.log("*** ajax success ***");
+            var jsonData2 = $.parseJSON(jsonData);
+            var laneFeatures = L.mapbox.featureLayer(jsonData2).addTo(app.activeMap);
+            app.state[region].laneLayers.push(laneFeatures);
+            laneFeatures.setStyle(app.state[region].bikeLaneStyle);
+            if ((app.display.loopCount < app.state[region].laneData.length - 1) && (app.state[region].laneData.length != 1)) {
+                console.log("multiple files");
+                app.display.loopCount++;
+                laneData = app.state[region].laneData[app.display.loopCount];
+                url = "bikelanes/" + laneData;
+                app.laneAjaxQueue(url, region);
+            } else {
+                console.log("single file");
+                app.display.loopCount = 0;
+                if (app.display.allRegions) {
+                    if (app.display.regionsArray.length > 0) {
+                        region = app.display.regionsArray[0];
+                        laneData = app.state[region].laneData[app.display.loopCount];
+                        app.display.regionsArray.shift();
+                        if (laneData) {
+                            url = "bikelanes/" + laneData;
+                            app.laneAjaxQueue(url, region);
+                        } else {
+                            console.log("=== ALERT: Missing data for", region);
+                        }
+                    } else {
+                        app.display.regionsArray = ["AL", "AR", "DC", "MO", "PG"];
+                    }
+                }
+            }
+        }).fail(function(){
+            console.log("*** ajax fail T ***");
+        });
+    },
+
+    // ======= bufferAjaxQueue =======
+    bufferAjaxQueue: function(url, region, buffer) {
+        console.log("bufferAjaxQueue");
+        $.ajax({
+            url: url,
+            method: "GET",
+            dataType: "text"
+        }).done(function(jsonData){
+            console.log("*** ajax success ***");
+            var jsonData2 = $.parseJSON(jsonData);
+            var bufferFeatures = L.mapbox.featureLayer(jsonData2).addTo(app.activeMap);
+            app.state[region].bufferLayers[buffer] = bufferFeatures;
+            bufferFeatures.setStyle(app.display.bufferStyle);
+        }).fail(function(){
+            console.log("*** ajax fail T ***");
```

Comment:
Need to add a user-friendly fail error in the UI instead of a developer-only `console.log` error eventually. This line can stay in this current PR for now.